hal/network/lwip/cellular/pppncpnetif.cpp

```diff
@@ -254,6 +254,24 @@ void PppNcpNetif::ncpEventHandlerCb(const NcpEvent& ev, void* ctx) {
         const auto& cev = static_cast<const CellularNcpAuthEvent&>(ev);
         LOG(TRACE, "New auth info");
         self->client_.setAuth(cev.user, cev.password);
+    } else if (ev.type == NcpEvent::POWER_STATE_CHANGED) {
+        const auto& cev = static_cast<const NcpPowerStateChangedEvent&>(ev);
+        if_event evt = {};
+        struct if_event_power_state ev_if_power_state = {};
+        evt.ev_len = sizeof(if_event);
+        evt.ev_type = IF_EVENT_POWER_STATE;
+        evt.ev_power_state = &ev_if_power_state;
+        if (cev.state == NcpPowerState::ON) {
+            evt.ev_power_state->state = IF_POWER_STATE_UP;
+            LOG(TRACE, "NCP power state changed: IF_POWER_STATE_UP");
+        } else if (cev.state == NcpPowerState::OFF) {
+            evt.ev_power_state->state = IF_POWER_STATE_DOWN;
+            LOG(TRACE, "NCP power state changed: IF_POWER_STATE_DOWN");
+        } else {
+            evt.ev_power_state->state = IF_POWER_STATE_NONE;
```

Comment:
Should we be generating such an event? What does it mean?

Reply:
We should notify the system power manager on startup to sync the modem state. On ESP32, we don't have a pin to indicate the modem state though.

Reply:
We do not use this for non-cellular interfaces though, right? ESP32 will also always be turned off on boot (see `esp32ncpnetif.cpp`).